game/core/src/me/lihq/game/screen/NavigationScreen.java

```diff
@@ -229,11 +235,17 @@ public void finishRoomTransition()
     @Override
     public void render(float delta)
     {
-
         game.player.pushCoordinatesToSprite();
+        for (AbstractPerson n : currentNPCS) {
+            n.pushCoordinatesToSprite();
+
+        }
 
         if (changeMap) {
             tiledMapRenderer.setMap(game.player.getRoom().getTiledMap());
+            tiledMapRenderer.clearPeople();
+            tiledMapRenderer.addPerson((List<AbstractPerson>) ((List<? extends AbstractPerson>) currentNPCS));
```

Comment:
I'll commit what I have, then we can revert if needed. So you can see the 2 options